go-controller/pkg/node/gateway_init.go

```diff
@@ -184,10 +184,19 @@ func getGatewayNextHops() ([]net.IP, string, error) {
 			}
 			gatewayIntf = defaultGatewayIntf
 		} else {
+			if gatewayIntf != defaultGatewayIntf {
```

Comment:
too much if's and else's for my taste :) but we can clean code later, I see value in printing different warnings, will help us debug better in the future.